src/index.js

```diff
@@ -140,44 +114,42 @@ class ReactCreditCards extends React.Component {
   }
 
   get options() {
-    const { number } = this.props;
-    const issuer = Payment.fns.cardType(number) || 'unknown';
+    const { issuer, number, preview } = this.props;
+    const { validCardTypes } = this.state;
+    let updatedIssuer = issuer || 'unknown';
+
+    if (number && !preview) {
+      const validatedIssuer = creditCardType(sanitizeNumber(number)).length ? creditCardType(sanitizeNumber(number))[0].type : 'unknown';
+
+      if (validCardTypes.includes(validatedIssuer)) {
+        updatedIssuer = validatedIssuer;
+      }
+    }
 
     let maxLength = 16;
 
-    if (issuer === 'amex') {
+    if (updatedIssuer === 'american-express') {
       maxLength = 15;
     }
-    else if (issuer === 'dinersclub') {
+    else if (updatedIssuer === 'diners-club') {
       maxLength = 14;
     }
-    else if (['hipercard', 'mastercard', 'visa'].includes(issuer)) {
+    else if (['hipercard', 'mastercard', 'visa'].includes(updatedIssuer)) {
       maxLength = 19;
     }
 
     return {
-      issuer,
+      issuer: updatedIssuer,
       maxLength,
     };
   }
 
   setCards() {
```

Comment:
So we need to update the `validCardTypes` if the `acceptedCards` prop changes after mount...
we can create a getter that would take the prop under consideration and return a memoized version.

Reply:
done